compiler/qsc_eval/src/val.rs

```diff
@@ -23,11 +23,68 @@ pub enum Value {
     Pauli(Pauli),
     Qubit(Qubit),
     Range(Option<i64>, i64, Option<i64>),
-    Result(bool),
+    Result(Result),
     String(Rc<str>),
     Tuple(Rc<[Value]>),
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum Result {
+    Val(bool),
+    Id(usize),
+}
+
+pub const RESULT_ZERO: Value = Value::Result(Result::Val(false));
+pub const RESULT_ONE: Value = Value::Result(Result::Val(true));
```

Comment:
I think it makes sense to put these inside `Value`. There's already `Value::unit`, these could be `Value::ZERO` and `Value::ONE` (the Result part is implied, IMO).

Reply:
Since `Value` can contain integers, big integers, and doubles too, `Value::ZERO()` feels ambiguous to me.

Reply:
what about something like:
```
    #[must_use]
    pub const fn result_zero() -> Self {
        Self::Result(Result::Val(false))
    }
```

Reply:
why did you change it from a const to a method?

Reply:
Because that's what `Value::unit` is, so I thought that was what you were suggesting. It's a `const` method at least, which `Value::unit` cannot be due to memory allocation.

Reply:
I was just suggesting to move the `const` inside the impl. `Value::unit` used to be a const `Value::UNIT` when it didn't need to allocate.